frontend/src/app/Settings/Users/UserFacilitiesSettings.tsx

```diff
@@ -0,0 +1,90 @@
+import React from "react";
+import { FieldErrors, UseFormRegister, UseFormSetValue } from "react-hook-form";
+
+import Checkboxes from "../../commonComponents/Checkboxes";
+import Required from "../../commonComponents/Required";
+
+import { UserFacilitySetting } from "./ManageUsersContainer";
+import "./ManageUsers.scss";
+import { CreateUser } from "./CreateUserSchema";
+import { alphabeticalFacilitySort } from "./UserFacilitiesSettingsForm";
+
+interface UserFacilitiesSettingProps {
+  formValues: CreateUser;
+  allFacilities: UserFacilitySetting[];
+  register: UseFormRegister<CreateUser>;
+  errors: FieldErrors<CreateUser>;
+  setValue: UseFormSetValue<CreateUser>;
+}
+
+// This is the react-hook-form supported version of UserFacilitiesSettingsForm.
+const UserFacilitiesSettings: React.FC<UserFacilitiesSettingProps> = ({
+  formValues,
+  allFacilities,
+  register,
+  errors,
+  setValue,
+}) => {
+  const isAdmin = formValues.role === "ADMIN";
+  const facilityAccessDescription = isAdmin
+    ? "Admins have access to all facilities"
+    : "All users must have access to at least one facility";
+  allFacilities.sort(alphabeticalFacilitySort);
+
+  const onChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value, checked } = e.target;
+    if (value === "ALL_FACILITIES") {
```

Comment:
nit: could we un-nest these logical checks and just put them all in the top row?